File: application-ver2/app/src/main/java/com/example/heartbeat/Temperature/RealTimeGraphTemp.java
package com.example.heartbeat.Temperature;

import android.graphics.Color;
import android.graphics.LinearGradient;
import android.graphics.Paint;
import android.util.Log;
import android.view.View;

import androidx.core.content.ContextCompat;

import java.math.*;

import com.example.heartbeat.R;
import com.github.mikephil.charting.charts.LineChart;
import com.github.mikephil.charting.components.YAxis;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.data.LineData;
import com.github.mikephil.charting.data.LineDataSet;
import com.github.mikephil.charting.interfaces.datasets.ILineDataSet;

public class RealTimeGraphTemp {
    private LineChart chart;
    private float minX;
    private float maxX;

    public RealTimeGraphTemp(View view){
        creatGraph(view);
    }

    private void creatGraph(View view){
        chart = (LineChart)view.findViewById(R.id.LineChart);
        chart.setDrawGridBackground(true);
        chart.setBackgroundColor(Color.WHITE);
        chart.setGridBackgroundColor(Color.WHITE);


        chart.getXAxis().setEnabled(false);

        YAxis leftAxis = chart.getAxisLeft();
        leftAxis.setEnabled(true);
        //leftAxis.setTextColor(getResources().getColor(R.color.colorGrid));
        leftAxis.setDrawGridLines(true);
        //leftAxis.setGridColor(getResources().getColor(R.color.colorGrid));


        YAxis rightAxis = chart.getAxisRight();
        rightAxis.setEnabled(false);

        minX = (float) 20.0;
        maxX = (float) 37.0;

        chart.getAxisLeft().setDrawLabels(true);
        chart.getAxisRight().setDrawLabels(false);
        chart.getXAxis().setDrawLabels(false);
        chart.getXAxis().setDrawGridLines(false);
        chart.getAxisLeft().setAxisMaximum(maxX);
        chart.getAxisLeft().setAxisMinimum(minX);
        chart.getDescription().setEnabled(false);

        Paint pain = chart.getRenderer().getPaintRender();

        chart.invalidate();


    }
    public void addEntry(double num) {
        LineData data = chart.getData();

        if(Math.abs((float)num - minX) > 2){
            minX = (float)num - (float) 1.8;
            chart.getAxisLeft().setAxisMinimum(minX);
        }

        if(Math.abs(maxX - (float)num) > 2){
            maxX = (float)num + (float) 1.8;
            chart.getAxisLeft().setAxisMaximum(maxX);
        }


        if (data == null) {
            data = new LineData();
            chart.setData(data);
        }

        ILineDataSet set = data.getDataSetByIndex(0);
        // set.addEntry(...); // can be called as well

        if (set == null) {
            set = createSet();
            data.addDataSet(set);
        }



        data.addEntry(new Entry((float)set.getEntryCount(), (float)num), 0);

        data.notifyDataChanged();

        // let the chart know it's data has changed
        chart.notifyDataSetChanged();

        chart.setVisibleXRangeMaximum(15);
        // this automatically refreshes the chart (calls invalidate())
        chart.moveViewTo(data.getEntryCount(), 50f, YAxis.AxisDependency.LEFT);

    }

    private LineDataSet createSet() {
        LineDataSet set = new LineDataSet(null, "체온");
        set.setDrawFilled(true);
        set.setFillColor(Color.RED);
        set.setLineWidth(1f);
        set.setDrawValues(false);
        set.setMode(LineDataSet.Mode.LINEAR);
        set.setDrawCircles(false);
        set.setHighLightColor(Color.RED);
        set.setColor(Color.RED);

        return set;
    }

}

File: Application_part/project_kotlin_version/Kotlin_ver/app/src/main/java/com/example/kotlin_ver/Temperature/RealTimeGraphTemp.kt
package com.example.kotlin_ver.Temperature

import android.graphics.Color
import android.view.View
import com.example.kotlin_ver.R
import com.github.mikephil.charting.charts.LineChart
import com.github.mikephil.charting.components.YAxis
import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.data.LineData
import com.github.mikephil.charting.data.LineDataSet
import com.github.mikephil.charting.interfaces.datasets.ILineDataSet

class RealTimeGraphTemp {
    private var chart: LineChart? = null
    private var minX = 0f
    private var maxX = 0f

    constructor(view: View) {
        creatGraph(view)
    }

    private fun creatGraph(view: View) {
        chart = view.findViewById<View>(R.id.LineChart) as LineChart
        chart!!.setDrawGridBackground(true)
        chart!!.setBackgroundColor(Color.WHITE)
        chart!!.setGridBackgroundColor(Color.WHITE)
        chart!!.xAxis.isEnabled = false
        val leftAxis = chart!!.axisLeft
        leftAxis.isEnabled = true
        leftAxis.setDrawGridLines(true)
        val rightAxis = chart!!.axisRight
        rightAxis.isEnabled = false
        chart!!.axisLeft.setDrawLabels(true)
        chart!!.axisRight.setDrawLabels(false)
        chart!!.xAxis.setDrawLabels(false)
        chart!!.xAxis.setDrawGridLines(false)
        chart!!.description.isEnabled = false
        val pain = chart!!.renderer.paintRender
        chart!!.invalidate()
    }

    fun addEntry(temp: TempData){
        var num = temp.value
        var data = chart!!.data
        if (Math.abs(num.toFloat() - minX) > 2) {
            minX = num.toFloat() - 1.8.toFloat()
            chart!!.axisLeft.axisMinimum = minX
        }
        if (Math.abs(maxX - num.toFloat()) > 2) {
            maxX = num.toFloat() + 1.8.toFloat()
            chart!!.axisLeft.axisMaximum = maxX
        }
        if (data == null) {
            data = LineData()
            chart!!.data = data
        }
        var set = data.getDataSetByIndex(0)
        // set.addEntry(...); // can be called as well
        if (set == null) {
            set = createSet()
            data.addDataSet(set)
        }
        data.addEntry(
            Entry(
                set.entryCount.toFloat(),
                num.toFloat()
            ), 0
        )
        data.notifyDataChanged()

        chart!!.notifyDataSetChanged()
        chart!!.setVisibleXRangeMaximum(15f)

        chart!!.moveViewTo(data.entryCount.toFloat(), 50f, YAxis.AxisDependency.LEFT)

    }


    private fun createSet(): LineDataSet {
        val set = LineDataSet(null, "체온")
        set.setDrawFilled(true)
        set.fillColor = Color.RED
        set.lineWidth = 1f
        set.setDrawValues(false)
        set.mode = LineDataSet.Mode.LINEAR
        set.setDrawCircles(false)
        set.highLightColor = Color.RED
        set.color = Color.RED
        return set
    }

}